codepy/libraries.py

```diff
@@ -123,18 +123,18 @@ def get_boost_libname(basename: str, aksetup: Config) -> list[str]:
 
 
 def add_boost_python(toolchain: Toolchain) -> None:
-    aksetup = get_aksetup_config()
     import sys
+
+    aksetup = get_aksetup_config()
+    version = sys.version_info[:2]
+
     toolchain.add_library(
             "boost-python",
             getlist(aksetup, "BOOST_INC_DIR", []),
             getlist(aksetup, "BOOST_LIB_DIR", []),
             [
-                *get_boost_libname("python-py{}{}".format(*sys.version_info[:2]),
-                                   aksetup),
-                "python{}.{}{}".format(
-                    *sys.version_info[:2],
-                    "m" if sys.version_info[0] >= 3 else "")
+                *get_boost_libname("python{}{}".format(*version), aksetup),
+                "python{}.{}{}".format(*version, sys.abiflags),
```

Comment:
I'm not sure this is correct. At least on my system
* The boost library is is called `libboost_python313.so`. It wasn't finding the previous one at all.
* Same for `libpython3.13.so`. As far as I can tell, the *m* flag is no longer there: https://docs.python.org/3/library/sys.html#sys.abiflags. I don't know if `abiflags` contains the `t` flag for the free-threaded version..

This seems to match my conda environment on `koelsch`, so it should be a bit more modern.

Reply:
Not sure there's a "correct" here. Debian used to do its own thing here, but I think they're now consistent with everyone else. "Works for someone" is good enough for me.